### ISSUE TITLE
SQL Planner threw 'NULL_TYPE is illegal in thrift stage' while trying to convert PlanFragment to thrift

### ISSUE THREAD

**Issue by wxl24life** [StarRocks/starrocks#29545]
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### Real behavior (Required)

```sh
java.lang.IllegalStateException: NULL_TYPE is illegal in thrift stage
        at com.google.common.base.Preconditions.checkState(Preconditions.java:512) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.analysis.Expr.treeToThriftHelper(Expr.java:752) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr.treeToThrift(Expr.java:730) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr.treesToThrift(Expr.java:336) ~[starrocks-fe.jar:?]
        at com.starrocks.planner.PlanFragment.toThrift(PlanFragment.java:326) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator$FragmentExecParams.toThriftForCommonParams(Coordinator.java:3244) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator$FragmentExecParams.toThriftInBatch(Coordinator.java:3451) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator.deliverExecBatchFragmentsRequests(Coordinator.java:1223) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator.deliverExecFragments(Coordinator.java:825) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator.exec(Coordinator.java:728) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.handleQueryStmt(StmtExecutor.java:756) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:446) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:334) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:451) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:709) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_362]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_362]
        at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_362]
```

### StarRocks version (Required)
- 2.5.8 throws error
- 2.3.2 runs ok

**Comment by ucasfl** [comment]
Should be fixed in https://github.com/StarRocks/starrocks/pull/24160.

**Comment by wxl24life** [comment]
confirmed, this issus has been fixed, thanks

> Should be fixed in #24160.
